WordPress/gutenberg
# Try adding dynamic page templates to pages section

## What?


Closes #50418.

Adds 404, Search and Homepage if it's set to posts, or Homepage and Blog page if home is set to a static page. 

Back buttons are now working (clicking back from a template accessed from the Pages list should go back to Pages). Would appreciate feedback on the implementation though as there might be a better way of doing it.

## Testing Instructions





1. Open site editor and navigate to Pages section
2. See templates displayed in pages list.
3. Click into a template, going back should return to Pages list.

### Testing Instructions for Keyboard


## Screenshots or screencast 
<img width="346" alt="Screenshot 2023-05-16 at 4 35 33 pm" src="https://github.com/WordPress/gutenberg/assets/8096000/5059e706-619d-42ac-b4bf-cbf4596563fc">
